src/viewer.js

```diff
@@ -143,6 +143,14 @@ export class Viewer {
     /** @const @private {!Object<string, string>} */
     this.params_ = {};
 
+    /** @private {?function(*)} */
+    this.onVisibilityResolve_ = null;
```

Comment:
"onVisibility" sounds ambiguous because it implies "visibility change". As opposed to "whenVisible" or such that makes it clear that it should yield once it becomes visible. Also, for consistency, if we are doing this logic for document-state in UserNotificationService, should we move this there as well?